session/session.go

```diff
@@ -312,32 +313,32 @@ func sanitizeBalInfo(balInfo perun.BalInfo) {
 	}
 }
 
-// retrieveParts retrieves the peers from corresponding to the aliases from the contacts provider.
+// retrieveParts retrieves the peer IDs corresponding to the aliases from the ID provider.
 // The order of entries for parts list will be same as that of aliases. i.e aliases[i] = parts[i].Alias.
-func retrieveParts(aliases []string, contacts perun.ContactsReader) ([]perun.Peer, error) {
-	knownParts := make(map[string]perun.Peer, len(aliases))
-	parts := make([]perun.Peer, len(aliases))
+func retrieveParts(aliases []string, idProvider perun.IDReader) ([]perun.PeerID, error) {
+	knownParts := make(map[string]perun.PeerID, len(aliases))
+	parts := make([]perun.PeerID, len(aliases))
 	missingParts := make([]string, 0, len(aliases))
 	repeatedParts := make([]string, 0, len(aliases))
 	foundOwnAlias := false
 	for idx, alias := range aliases {
 		if alias == perun.OwnAlias {
 			foundOwnAlias = true
 		}
-		peer, isPresent := contacts.ReadByAlias(alias)
+		peerID, isPresent := idProvider.ReadByAlias(alias)
 		if !isPresent {
 			missingParts = append(missingParts, alias)
 			continue
 		}
 		if _, isPresent := knownParts[alias]; isPresent {
 			repeatedParts = append(repeatedParts, alias)
 		}
-		knownParts[alias] = peer
-		parts[idx] = peer
+		knownParts[alias] = peerID
+		parts[idx] = peerID
 	}
 
 	if len(missingParts) != 0 {
```

Comment:
```suggestion
	if len(missingPartIDs) != 0 {
```

Reply:
@anagha-ks As discussed, we will rename only `parts` to `partIDs` while leaving the other variables (`missingParts`, `repeatedParts`, `knownParts`, Because the former variable refers to a list of `peerID` while the latter variables  refer to either a list of aliases or off-chain addresses.

Hence I have dropped all the above suggestions relating to this by resolving them.